### DISCUSSION TITLE
Enforcing standards via eslint

### DISCUSSION THREAD

**Discussion by cafesanu** [TanStack/query#3992] [Ideas] [answered]
Hello! First, let me just thank you for `react-query`. We love it!

Now, our team is in the process of migrating to react query and we want to define and enforce standards. We usually do it via plugins, but I find that the current plugin https://github.com/KubaJastrz/eslint-plugin-react-query is a bit stale and probably not well maintained.

Have you though about creating an official eslint rules plugin, as part of the `TanStack` repo? This way, teams can decide the basic standards?

Thanks!

**Comment by TkDodo** [comment]
I've never seen this eslint plugin, but it looks interesting. Especially the `prefer-query-object-syntax` is in-line with what we want to communicate, as we'd like to get rid of the overloads eventually.

Not sure if you know the maintainers, but now that we have the TanStack mono-repo, we could actually distribute the eslint-plugin directly from the TanStack.

I don't agree with some of the rules though (like mutations needing a key as recommended),but I think we can discuss this.

**Reply by cafesanu** [reply]
No idea who the owners are. I think that library is stale anyway, so not sure if it's worth to merge it or start from scratch

**Comment by cafesanu** [comment, marked as answer]
@TkDodo should this be closed now that we have the eslint plugin? 🎉🎉 https://tanstack.com/query/latest/docs/react/eslint/eslint-plugin-query
